src/runtime/local/vectorized/MTWrapper_dense.cpp

```diff
@@ -31,11 +31,32 @@ void MTWrapper<DenseMatrix<VT>>::executeSingleQueue(
     mem_required += this->allocateOutput(res, numOutputs, outRows, outCols, combines);
     auto row_mem = mem_required / len;
 
-    // create task queue (w/o size-based blocking)
-    std::unique_ptr<TaskQueue> q = std::make_unique<BlockingTaskQueue>(len);
+    std::vector<std::unique_ptr<TaskQueue>> q;
+    std::vector<TaskQueue*> qvector;
+    if (ctx->getUserConfig().pinWorkers) {
+        for(int i=0; i<this->_numQueues; i++) {
+            // Ideally queues would be created as such:
+            // q.emplace_back(new BlockingTaskQueue(len));
+            // However I'm getting an error so they are created with make_unique and get now.
```

Comment:
The BlockingTaskQueue class contains a non copy-able/moveable member (the condition variable). This is why you have to work with (unique) pointers here.